File: 剑指Offer/39.数组中出现超过一半的数字.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        preNum = nums[0]
        count = 1
        for i in range(1, len(nums)):
            if nums[i] == preNum:
                count += 1
            else:
                count -= 1

            if count < 0:
                preNum = nums[i]
                count = 1
        return preNum

File: 剑指Offer/39.数组中出现超过一半的数字.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        int preNum = nums[0];
        int count = 1;
        for (int i = 1; i < nums.size(); ++i) {
            if (nums[i] == preNum) {
                ++count;
            }
            else{
                --count;
            }

            if (count < 0) {
                preNum = nums[i];
                count = 1;
            }
        }
        return preNum;
    }
};